pkg/operator/ceph/cluster/controller.go

```diff
@@ -373,8 +373,9 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 		for _, daemon := range monitorDaemonList {
 			isDisabled = isMonitoringDisabled(daemon, cluster.Spec)
 			if _, ok := cluster.monitoringChannels[daemon]; ok {
-				if !isDisabled {
+				if !isDisabled && cluster.monitoringChannels[daemon].monitoringRunning {
```

Comment:
Do we even need to check if `isDisabled`? Seems like we should always ensure the monitoring is stopped to avoid a bug that leaves the monitoring running. 
```suggestion
				if monitoring.monitoringRunning {
```